### PR TITLE
Dock Construction Updates

### PR DESCRIPTION
**Pull request by Beatdotroot**
#### Summary
Bugfixes "Allows certain docks to be placed on non-flowing shallow and deep water."

#### Description
#61051

#### Describe the solution
Adds 2 new dock construction options in "build_deep_river_dock" and "build_river_dock_shallow_bridge" groups

#### Describe alternatives you've considered
Unsure.

#### Testing
Creating a testing world, spawned resources with debug menu, and built multiple bridges.

#### Additional context
N/A

### CODE REVIEW

**Comment by ashGlaw** [comment]
The construction menu needs so much attention. Thank you for giving it some love <3

**Review by MNG-cataclysm — APPROVED** [review]

**Comment by Beatdotroot** [comment]
Of course, thank you!!

**Comment by Fris0uman** [comment]
Did you check that changing the ids of the recipe doesn't affect save made before that change? Also does this lead to the magical situation where I'm digging fundation for support deep underwater with just a shovel and a hammer from the confort of any floating vehicle?

**Comment by Beatdotroot** [comment]
I actually had made it with the express purpose of using it on a prior save, so it definitely works with saves made before the change. I also believe that second issue was technically already a thing you could do in a vehicle floating in a river, just not a lake.

File: data/json/construction.json [hunk 1]
```diff
@@ -4820,7 +4820,7 @@
   },
   {
     "type": "construction",
-    "id": "constr_dock",
+    "id": "constr_dock_moving",
```
Comment (Maleclypse) [inline data/json/construction.json:4823]:
```suggestion
    "id": "constr_dock_moving",/
```

File: data/json/construction.json [hunk 2]
```diff
@@ -4820,7 +4820,7 @@
   },
   {
     "type": "construction",
-    "id": "constr_dock",
+    "id": "constr_dock_moving",/
```
Comment (Maleclypse) [inline data/json/construction.json:4823]:
```suggestion
    "id": "constr_dock_moving",
```

**Pull request merged**